### ISSUE TITLE
Arbitrary upper limit to code length?

### ISSUE THREAD

**Issue by jtomazin** [ThinkAlexandria/BoringAuth#2]
I was trying to generate larger TOTP values and noticed that [output values are capped at 2^31 bits](https://github.com/ThinkAlexandria/BoringAuth/blob/f1ddeb031cb90d381319039e021b1e4f0d3c6327/src/oath/totp.rs#L270) -- is there a reason for this? The [HOTP paper](https://tools.ietf.org/html/rfc4226) recommends that an easy way to improve security is to just extract more characters from the HMAC hash.

**Comment by YetAnotherMinion** [comment]
You are correct. I believe it makes sense to allow extracting all of the bits available in the HMAC hash. So if you are using sha256 you can extract up to 24 characters, if you use sha512 you could extract up to 48 characters.

@jtomazin Would this work for you.
